extension/lib/shopgate/CartExtensionPipeline.js

```diff
@@ -49,7 +49,45 @@ class ShopgateCartExtensionPipeline {
       await this._bigCommerceCartRepository.addItems(itemsToAdd)
     }
     if (itemsToUpdate.length) {
-      await this._bigCommerceCartRepository.updateItems(itemsToUpdate, () => {})
+      await this.updateItems(itemsToUpdate, () => {}, true)
+    }
+  }
+
+  /**
+  * @param {[BigCommerceCartLineItemUpdateRequest]} itemsToUpdate
+  * @param {bigCommerceUpdateFailureNotifier} updateFailureNotifier
+  * @param {boolean} handleError
+  * @return {Promise<void>}
+  */
+  async updateItems (itemsToUpdate, updateFailureNotifier, handleError) {
+    const cartId = await this.getCartId()
+    try {
+      await this._bigCommerceCartRepository.updateItems(itemsToUpdate, updateFailureNotifier)
+    } catch (err) {
+      if (err.code === 422) {
+        const errorMessageMatch = err.message.match(/({.+})/)
+        let errorMessage = null
+        if (errorMessageMatch) {
+          try {
+            errorMessage = JSON.parse(errorMessageMatch[1]).title
+          } catch (err) {
+            errorMessage = 'Items in your cart couldn\'t be updated. Please try again later.'
+            this._context.log.error(decorateError(err), 'Unable to process the error from BigC api')
+          }
+        }
+        let ecartError = new Error('Failed adding products to cart')
+        ecartError.code = 'ECART'
+        ecartError.errors = [{
+          code: 'NOTAVAILABLE',
+          message: errorMessage,
+          translated: false
+        }]
+
+        await this._messagesRepository.push(cartId, errorMessage)
+        if (handleError) throw ecartError
+      }
+      await this._messagesRepository.push(cartId, 'We were unable to update the cart. Please try again later.')
```

Comment:
you add both errors into this message repo, either add into else statement or return early out of 422 check